gnomad_constraint/utils/constraint.py

```diff
@@ -391,6 +398,208 @@ def apply_models(
+
+    if not summary_file:
+        summary_file = new_temp_file(prefix="constraint", extension="he")
+
+    all_possible_dtype = count_variants_by_group(
+        context_ht,
+        omit_methylation=omit_methylation,
+        return_type_only=True,
+    ).variant_count.dtype
+
+    # Count possible variants in context Table.
+    if recalculate_all_possible_summary:
+        all_possible = count_variants_by_group(
+            context_ht,
+            omit_methylation=omit_methylation,
+        ).variant_count
+        # with hl.hadoop_open(summary_file, "wb") as f:
+        #     pickle.dump(all_possible, f)
+        hl.experimental.write_expression(all_possible, summary_file)
+    # with hl.hadoop_open(summary_file, "rb") as f:
+    #     all_possible = pickle.load(f)
+    all_possible = hl.experimental.read_expression(summary_file)
+
+    if recalculate_all_possible_summary_unfiltered:
+        all_possible_unfiltered = count_variants_by_group(
+            raw_context_ht.rows(), omit_methylation=omit_methylation
+        ).variant_count
+        all_possible_unfiltered = {
+            x: y
+            for x, y in list(all_possible_unfiltered.items())
+            if x.context is not None
+        }
+        hl.experimental.write_expression(
+            all_possible_unfiltered, new_temp_file(prefix="constraint", extension="he")
+        )
+        # with hl.hadoop_open(new_temp_file(prefix="constraint", extension="he"), "wb") as f:
+        #     pickle.dump(all_possible_unfiltered, f)
+    # Uncomment this line and next few commented lines to back-calculate total_mu from the old mutation rate dataset
+    # all_possible_unfiltered = load_all_possible_summary(filtered=False)
+
+    # Count the observed variants in the genome sites Table.
+    genome_ht = count_variants_by_group(
+        genome_ht,
+        count_downsamplings=pops,
+        count_singletons=count_singletons,
+        omit_methylation=omit_methylation,
+    )
+
+    old_mu_data = mutation_rate_ht.ht()
+    ht = genome_ht.annotate(
+        possible_variants=hl.literal(all_possible, dtype=all_possible_dtype)[
+            genome_ht.key
+        ],
+        # possible_variants_unfiltered=hl.literal(all_possible_unfiltered, dtype=all_possible_dtype)[genome_ht.key],
+        old_mu_snp=old_mu_data[
+            hl.struct(context=genome_ht.context, ref=genome_ht.ref, alt=genome_ht.alt)
+        ].mu_snp,
+    )
+
+    ht = ht.checkpoint(new_temp_file(prefix="constraint", extension="ht"))
+
+    total_bases = ht.aggregate(hl.agg.sum(ht.possible_variants)) // 3
+    # total_bases_unfiltered = ht.aggregate(hl.agg.sum(ht.possible_variants_unfiltered)) // 3
+    # total_mu = ht.aggregate(hl.agg.sum(ht.old_mu_snp * ht.possible_variants_unfiltered) / total_bases_unfiltered)
+
+    # Get the index of dowsamplings with size of 1000 genomes
+    downsamplings = list(map(lambda x: x[1], get_downsamplings(ht.freq_meta)))
+    downsampling_idx = downsamplings.index(1000)
```

Comment:
This is used for all the pops, but obtained for only global, will this for sure be the same index for other pops as it is for global?

```suggestion
    # Get the index of dowsampling with size of `downsampling_level`.
    downsampling_idx = downsampling_meta.map(lambda x: hl.int(x[1]['downsampling'])).index(downsampling_level)
```